### DISCUSSION TITLE
Need 'correct' method of getting usb device number in monitor function

### DISCUSSION THREAD

**Discussion by Eqqm4n** [pyudev/pyudev#494] [Q&A]
So my monitor function is as follows ::
`def log_usb_event(action, device) :`
`    if (action == "remove") and (device is not None) :`
`        properties = device.properties`
`        for p in properties :`
`            if p == "DEVNUM" :`
`                address = int(device.__getitem__(p))`

My goal here is to get the USB device number assigned by the kernel and check it against the list of devices I maintain so I know when something was removed (rather than constantly generating updated lists and purging by hand).  I should note that this code works, however, when I unplug a device for the first time I get this warning after executing the `address = int(device.__getitem__(p))` line ::

`DeprecationWarning: Will be removed in 1.0. Access properties with Device.properties`

What is then the preferred way to execute this code?  From what I can tell in the manpages `Device.properties` gives a list of what properties a device has, but does not include what the values of those properties are.

**Comment by mulkieran** [comment]
I think this is just a Python confusion.

Will
```
for name, value in device.properties.items():
    if name == "DEVNUM":
        address = value
```
do it?
